wasm/jit/testdata/unreachable.wat

```diff
@@ -0,0 +1,14 @@
+(module
+	(func $cause_unreachable (export "cause_unreachable")
+		(call $one)
+	)
+	(func $one
+		(call $two)
+	)
+	(func $two
+		(call $three)
+	)
+	(func $three
+		(unreachable)
```

Comment:
is there a more intuitive use of unreachable? like something where it is may occur in real life (not a forged example)

Reply:
for example `unreachable` will be placed in a branch for the case when the slice boundary check failed (like TinyGO does emit unreachable on "slice out of bounds")!

Reply:
But it is essentially the same as this example